modules/hostap/src/supp_api.c

```diff
@@ -344,13 +364,22 @@ int z_wpa_supplicant_connect(const struct device *dev,
 	_wpa_cli_cmd_v("enable_network %d", resp.network_id);
 
 	if (params->channel != WIFI_CHANNEL_ANY) {
-		int freq = chan_to_freq(params->channel);
+		int freq;
 
-		if (freq < 0) {
-			ret = -1;
-			wpa_printf(MSG_ERROR, "Invalid channel %d",
-				params->channel);
-			goto out;
+		if (params->band != WIFI_FREQ_BAND_UNKNOWN) {
+			freq = wpa_supp_band_chan_compat(wpa_s, params->band, params->channel);
+			if (freq < 0) {
+				_wpa_cli_cmd_v("remove_network %d", resp.network_id);
+				goto out;
+			}
+		} else {
+			freq = chan_to_freq(params->channel);
+			if (freq < 0) {
+				ret = -1;
```

Comment:
Since we are returning error in this case, I think we can remove the network as is being done in case of valid band.

Reply:
Moved the cleanup to goto to handle multiple exits.